packages/telegram-stamper/src/index.ts

```diff
@@ -0,0 +1,185 @@
+/// <reference lib="dom" />
+import { ApiKeyStamper } from "@turnkey/api-key-stamper";
+import { TelegramStamperError } from "./errors";
+
+declare global {
+  interface Window {
+    Telegram: any;
+  }
+}
+
+// config when telegram stamper is passed in an api key
+export type TTelegramStamperConfig = {
+  apiPublicKey: string;
+  apiPrivateKey: string;
+  cloudStorageKeySuffix?: string;
+};
+
+type CloudStorageAPIKey = {
+  apiPublicKey: string;
+  apiPrivateKey: string;
+};
+
+// Constant for default key name, note this should NOT be used directly, use the getCloudStorageKeyName() function as it includes the user passed key suffix
+const TURNKEY_CLOUD_STORAGE_KEY = "TURNKEY_API_KEY";
+
+/**
+ * Stamper to use with `@turnkey/http`'s `TurnkeyClient`
+ */
+export default class TelegramStamper {
+  stamper: ApiKeyStamper | undefined;
+  cloudStorageKeySuffix: string = "";
```

Comment:
The fact that a stamper contains another stamper might be a bit surprising for some readers. I wonder if that's worth a comment in the top level docstring?

We could call this `TelegramCloudStorageStamperContainer` if we were writing enterprise Java...but we're not. So I'm down to keep `TelegramStamper` or `CloudStorageStamper` (slight preference because it's more specific and describes the implementation better?)

Reply:
changed to TelegramCloudStorageStamper